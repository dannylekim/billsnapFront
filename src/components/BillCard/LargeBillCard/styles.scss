@import "../../../variables";

div.large__bill {
  height: 600px;
  width: 80%;
  margin-left: auto;
  margin-right: 32px;
  margin-top: 50px;
  background: $card-color;
  box-shadow: 0 7.5px 60px rgba(90, 97, 105, 0.25);
  border-radius: 10px;

  font-family: $primary-font;
  font-style: normal;
  font-weight: 500;
  font-size: 25px;
  line-height: 45px;
  color: $primary-text-color;
}

div.inner__bill_container {
  height: 81.09%;
  width: 87.1%;
  margin-left: auto;
  margin-right: auto;
  margin-top: 40px;
  border: 1px solid rgba(0, 0, 0, 0.59);
  box-sizing: border-box;
  border-radius: 26px;
}
