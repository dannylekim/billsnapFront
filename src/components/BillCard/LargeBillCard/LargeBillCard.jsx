import React from "react";
import "./styles.scss";
import {Card, CardBody} from "shards-react";
import AccountStatusIcon from "../../AccountStatusIcon";
import ItemSplitContainer from "../../ItemSplitContainer";

function getItems(bill) {
  return (
    <>
      {bill.items.map((item) => (
        <div className="split_between embolden" key={item.id}>
          <div>
            <div className="split_between">
              <div className="sub_items_text">{item.name}</div>
              <ItemSplitContainer bill={bill} selectedItemId={item.id} />
            </div>

          </div>
          <div>{item.cost} $</div>
        </div>
      ))}
    </>
  );
}

const LargeBillCard = ({ bill }) => {
  const itemCost = bill.items
    .map((item) => item.cost)
    .map((cost) => parseFloat(cost))
    .reduce((accumulator, currentValue) => accumulator + currentValue);

  return (
    <Card className="large__bill">
      <div className="inner__bill__dimensions">
        <div className="split_between space-top">
          <div>Bill # {bill.id}</div>
          <div className="embolden">{bill.name}</div>
        </div>
        <div className="inner__bill_container">
          <CardBody>
            <div className="split_between">
              <div />
              <div className="sub-text">
                {bill.created.toLocaleString("en-AU")}
              </div>
            </div>
            <div className="split_between">
              <div />
              <div className="sub-text">
                Category: {bill.category ? bill.category : "None"}
              </div>
            </div>

            <br />
            <div className="split_between">
              <div className="sub-text">Items:</div>
              <div />
            </div>
            {getItems(bill)}
            <div className="split_between">
              <div className="sub-text">Tips:</div>
              <div className="embolden">{bill.totalTip} $</div>
            </div>
            <div className="split_between">
              <div className="sub-text">Taxes:</div>
              {bill.taxes.length === 0 ? (
                <div className="embolden"> 0.00 $</div>
              ) : (
                <div className="embolden">
                  {(bill.balance - itemCost - bill.totalTip).toFixed(2)} $
                </div>
              )}
            </div>
            <div className="split_between embolden total_balance">
              <div>Total Balance</div>
              <div>{bill.balance} $</div>
            </div>
            <div className="account__icons__container">
              {bill.informationPerAccount.map((accountInfo, key) => (
                <AccountStatusIcon
                  key={key}
                  name={accountInfo.account.firstName}
                  status={accountInfo.invitationStatus}
                />
              ))}
            </div>
          </CardBody>
        </div>
      </div>
    </Card>
  );
};
export default LargeBillCard;
