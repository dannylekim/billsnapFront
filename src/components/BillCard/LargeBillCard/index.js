import LargeBillCard from './LargeBillCard.jsx';
export default LargeBillCard;