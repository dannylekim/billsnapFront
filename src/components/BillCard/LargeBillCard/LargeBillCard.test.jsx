import React from "react";
import LargeBillCard from "./LargeBillCard";

jest.mock("../../../components/ItemSplitContainer", () => "ItemSplitContainer");

const sampleBill = {
  id: 100,
  name: "long bill",
  creator: {
    firstName: "string",
    middleName: "string",
    lastName: "string",
    gender: "MALE",
    phoneNumber: "string",
    birthDate: "2020-07-25",
    location: {
      name: "string",
      description: "string",
      address: "string",
      city: "string",
      state: "string",
      country: "string",
      postalCode: "string",
    },
    id: 23,
    email: "test@Password123.com",
  },
  responsible: {
    firstName: "string",
    middleName: "string",
    lastName: "string",
    gender: "MALE",
    phoneNumber: "string",
    birthDate: "2020-07-25",
    location: {
      name: "string",
      description: "string",
      address: "string",
      city: "string",
      state: "string",
      country: "string",
      postalCode: "string",
    },
    id: 23,
    email: "test@Password123.com",
  },
  status: "OPEN",
  company: "LONG",
  category: "shopping",
  created: "28-08-2020 16:13:46 -0400",
  updated: "28-08-2020 16:13:46 -0400",
  items: [
    {
      id: 116,
      name: "item 6",
      cost: "6.00",
    },
    {
      id: 121,
      name: "item 3",
      cost: "3.00",
    },
    {
      id: 118,
      name: "item 2",
      cost: "2.00",
    },
    {
      id: 120,
      name: "item 4",
      cost: "4.00",
    },
    {
      id: 117,
      name: "item 5",
      cost: "5.00",
    },
    {
      id: 119,
      name: "item 1",
      cost: "1.00",
    },
  ],
  informationPerAccount: [
    {
      account: {
        firstName: "string",
        middleName: "string",
        lastName: "string",
        gender: "MALE",
        phoneNumber: "string",
        birthDate: "2020-07-25",
        location: {
          name: "string",
          description: "string",
          address: "string",
          city: "string",
          state: "string",
          country: "string",
          postalCode: "string",
        },
        id: 53,
        email: "many@bills.com",
      },
      items: [],
      subTotal: "0.00",
      tip: "0.00",
      taxes: "0.00",
      amountPaid: "0.00",
      amountRemaining: "0.00",
      invitationStatus: "PENDING",
      paidStatus: null,
      total: "0.00",
    },
    {
      account: {
        firstName: "string",
        middleName: "string",
        lastName: "string",
        gender: "MALE",
        phoneNumber: "string",
        birthDate: "2020-07-25",
        location: {
          name: "string",
          description: "string",
          address: "string",
          city: "string",
          state: "string",
          country: "string",
          postalCode: "string",
        },
        id: 23,
        email: "test@Password123.com",
      },
      items: [
        {
          itemId: 116,
          name: "item 6",
          cost: 6,
          percentage: 100,
        },
        {
          itemId: 121,
          name: "item 3",
          cost: 3,
          percentage: 100,
        },
        {
          itemId: 118,
          name: "item 2",
          cost: 2,
          percentage: 100,
        },
        {
          itemId: 120,
          name: "item 4",
          cost: 4,
          percentage: 100,
        },
        {
          itemId: 117,
          name: "item 5",
          cost: 5,
          percentage: 100,
        },
        {
          itemId: 119,
          name: "item 1",
          cost: 1,
          percentage: 100,
        },
      ],
      subTotal: "21.00",
      tip: "69.00",
      taxes: "54.68",
      amountPaid: "0.00",
      amountRemaining: "144.68",
      invitationStatus: "ACCEPTED",
      paidStatus: null,
      total: "144.68",
    },
  ],
  splitBy: "ITEM",
  totalTip: "69.00",
  balance: "144.68",
  taxes: [
    {
      name: "tax 3",
      percentage: 30,
      id: 20,
    },
    {
      name: "tax 4",
      percentage: 40,
      id: 21,
    },
    {
      name: "tax 2",
      percentage: 20,
      id: 19,
    },
    {
      name: "tax 5",
      percentage: 50,
      id: 22,
    },
    {
      name: "tax 1",
      percentage: 10,
      id: 23,
    },
  ],
};

describe("LargeBillCard", () => {
  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("SmallBillCard should match snap shot", () => {
        matches(<LargeBillCard bill={sampleBill} />);
      });
    });
  });
});
