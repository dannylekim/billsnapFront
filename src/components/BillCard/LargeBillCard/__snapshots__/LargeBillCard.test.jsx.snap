// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeBillCard render snapshots ðŸ“¸ SmallBillCard should match snap shot 1`] = `
<div
  className="large__bill card"
>
  <div
    className="inner__bill__dimensions"
  >
    <div
      className="split_between space-top"
    >
      <div>
        Bill # 
        100
      </div>
      <div
        className="embolden"
      >
        long bill
      </div>
    </div>
    <div
      className="inner__bill_container"
    >
      <div
        className="card-body"
      >
        <div
          className="split_between"
        >
          <div />
          <div
            className="sub-text"
          >
            28-08-2020 16:13:46 -0400
          </div>
        </div>
        <div
          className="split_between"
        >
          <div />
          <div
            className="sub-text"
          >
            Category: 
            shopping
          </div>
        </div>
        <br />
        <div
          className="split_between"
        >
          <div
            className="sub-text"
          >
            Items:
          </div>
          <div />
        </div>
        <div
          className="split_between embolden"
        >
          <div>
            <div
              className="split_between"
            >
              <div
                className="sub_items_text"
              >
                item 6
              </div>
              <div
                className="popout__button"
              >
                <svg
                  fill="currentColor"
                  height={15}
                  onClick={[Function]}
                  size={15}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            6.00
             $
          </div>
        </div>
        <div
          className="split_between embolden"
        >
          <div>
            <div
              className="split_between"
            >
              <div
                className="sub_items_text"
              >
                item 3
              </div>
              <div
                className="popout__button"
              >
                <svg
                  fill="currentColor"
                  height={15}
                  onClick={[Function]}
                  size={15}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            3.00
             $
          </div>
        </div>
        <div
          className="split_between embolden"
        >
          <div>
            <div
              className="split_between"
            >
              <div
                className="sub_items_text"
              >
                item 2
              </div>
              <div
                className="popout__button"
              >
                <svg
                  fill="currentColor"
                  height={15}
                  onClick={[Function]}
                  size={15}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            2.00
             $
          </div>
        </div>
        <div
          className="split_between embolden"
        >
          <div>
            <div
              className="split_between"
            >
              <div
                className="sub_items_text"
              >
                item 4
              </div>
              <div
                className="popout__button"
              >
                <svg
                  fill="currentColor"
                  height={15}
                  onClick={[Function]}
                  size={15}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            4.00
             $
          </div>
        </div>
        <div
          className="split_between embolden"
        >
          <div>
            <div
              className="split_between"
            >
              <div
                className="sub_items_text"
              >
                item 5
              </div>
              <div
                className="popout__button"
              >
                <svg
                  fill="currentColor"
                  height={15}
                  onClick={[Function]}
                  size={15}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            5.00
             $
          </div>
        </div>
        <div
          className="split_between embolden"
        >
          <div>
            <div
              className="split_between"
            >
              <div
                className="sub_items_text"
              >
                item 1
              </div>
              <div
                className="popout__button"
              >
                <svg
                  fill="currentColor"
                  height={15}
                  onClick={[Function]}
                  size={15}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            1.00
             $
          </div>
        </div>
        <div
          className="split_between"
        >
          <div
            className="sub-text"
          >
            Tips:
          </div>
          <div
            className="embolden"
          >
            69.00
             $
          </div>
        </div>
        <div
          className="split_between"
        >
          <div
            className="sub-text"
          >
            Taxes:
          </div>
          <div
            className="embolden"
          >
            54.68
             $
          </div>
        </div>
        <div
          className="split_between embolden total_balance"
        >
          <div>
            Total Balance
          </div>
          <div>
            144.68
             $
          </div>
        </div>
        <div
          className="account__icons__container"
        >
          <div>
            <div
              className="accounts__icons"
            >
              <svg
                fill="currentColor"
                height={50}
                size={50}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width={50}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              <svg
                className="status_icon"
                color="#FFB400"
                fill="currentColor"
                height={30}
                size={30}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": "#FFB400",
                  }
                }
                viewBox="0 0 24 24"
                width={30}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 2v6h.01L6 8.01 10 12l-4 4 .01.01H6V22h12v-5.99h-.01L18 16l-4-4 4-3.99-.01-.01H18V2H6z"
                />
              </svg>
            </div>
            <p
              className="icon__name"
            >
              string
            </p>
          </div>
          <div>
            <div
              className="accounts__icons"
            >
              <svg
                fill="currentColor"
                height={50}
                size={50}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width={50}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              <svg
                className="status_icon"
                color="#47E5B6"
                fill="currentColor"
                height={40}
                size={40}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": "#47E5B6",
                  }
                }
                viewBox="0 0 24 24"
                width={40}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                />
              </svg>
            </div>
            <p
              className="icon__name"
            >
              string
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
